File: 프로그래머스/lv1/68644. 두 개 뽑아서 더하기/두 개 뽑아서 더하기.js
// function solution(numbers) {
//   let answer = [];

//   for (let i = 0; i < numbers.length - 1; i++) {
//     for (let j = i + 1; j < numbers.length; j++) {
//       tmpSum = numbers[i] + numbers[j];
//       if (answer.indexOf(tmpSum) === -1) {
//         answer.push(tmpSum);
//       }
//     }
//   }
//   answer.sort((a, b) => a - b);

//   return answer;
// }
function solution(numbers) {
  let answer = new Set() // add로 +

  // 반복문
  for (let i=0; i<numbers.length-1; i++) {
    for (let j=i+1; j<numbers.length; j++) {
      let tmpSum = numbers[i] + numbers[j]
      answer.add(tmpSum)
    }
  }


  return [...answer].sort((a,b)=>a-b)
}

File: 프로그래머스/1/68644. 두 개 뽑아서 더하기/두 개 뽑아서 더하기.java
import java.util.ArrayList;
import java.util.Arrays;
import java.util.Collections;

class Solution {
    public int[] solution(int[] numbers) {
        ArrayList<Integer> list = new ArrayList<Integer>();
        
        for (int i=0; i<numbers.length; i++) {
            for (int j=i+1; j<numbers.length; j++) {
                int tmpSum = numbers[i]+numbers[j];
                if (list.indexOf(tmpSum) < 0) {
                    list.add(tmpSum);
                }
            }
        }
        Collections.sort(list);
        
        // 정렬된 list를 배열로 변환
        int[] result = new int[list.size()];
        for (int i=0; i<list.size(); i++) {
            result[i] = list.get(i);
        }
        
        return result;
    }
}